src/reweighting.hxx

```diff
@@ -92,4 +93,52 @@ auto topptreweighting(auto &df, const std::string &weightname,
                          {gen_pdgids, gen_status, gen_pt});
     return df1;
 }
+
+/**
+ * @brief Function used to evaluate Z pt mass weights
+ *
+ * @param df The input dataframe
+ * @param weightname name of the generated weight
+ * @param gen_boson name of the column that contains a pair of Lorentzvectors,
+ * where the first one is the one of the genboson
+ * @param workspace_file path to the file which contains the workspace to be
+ * read
+ * @param functor_name name of the function from the workspace
+ * @param argset arguments of the function
+ * @return a new dataframe containing the new column
+ */
+auto zPtMassReweighting(auto &df, const std::string &weightname,
+                        const std::string &gen_boson,
+                        const std::string &workspace_file,
+                        const std::string &functor_name,
+                        const std::string &argset) {
+
+    // retrieve pt and mass of gen boson
+    auto df1 = df.Define(gen_boson + "_pt",
```

Comment:
I would be in favour of putting this and the define below into seperate functions. This way it would also be possible to writeout genboson pt and gensobon mass as quantities.